src/zenml/integrations/mlflow/mlflow_environment.py

```diff
@@ -98,3 +107,107 @@ def deactivate(self) -> None:
     def tracking_uri(self) -> str:
         """Returns the MLflow tracking URI for the current stack."""
         return self._mlflow_tracking_uri
+
+
+class MLFlowStepEnvironment(BaseEnvironmentComponent):
+    """Provides information about an MLflow step environment.
+    To access it inside your step function:
+
+    ```python
+    from zenml.environment import Environment
+    from zenml.integrations.mlflow.mlflow_environment import MLFLOW_STEP_ENVIRONMENT_NAME
+
+
+    @step
+    def my_step(...)
+        env = Environment[MLFLOW_STEP_ENVIRONMENT_NAME]
+        do_something_with(env.mlflow_tracking_uri)
+    ```
+
+    """
+
+    NAME = MLFLOW_STEP_ENVIRONMENT_NAME
+
+    def __init__(self, experiment_name: str, run_name: str):
+        """Initialize a MLflow step environment component.
+
+        Args:
+            experiment_name: the experiment name under which all MLflow
+                artifacts logged under the current step will be tracked.
+                If no MLflow experiment with this name exists, one will
+                be created when the environment component is activated.
+            run_name: the name of the MLflow run associated with the current
+                step. If a run with this name does not exist, one will be
+                created when the environment component is activated,
+                otherwise the existing run will be reused.
+        """
+        super().__init__()
+        self._experiment_name = experiment_name
+        self._experiment = None
+        self._run_name = run_name
+        self._run = None
+
+    def _create_or_reuse_mlflow_run(self) -> None:
+        """Create or reuse an MLflow run for the current step.
+
+        IMPORTANT: this function is not race condition proof. If two or more
```

Comment:
```suggestion
        IMPORTANT: this function might cause a race condition. If two or more
```

Possibly a bit clearer? Feel free to ignore this one.